source/slang/slang-capabilities.capdef

```diff
@@ -587,8 +585,8 @@ alias DX_6_7 = sm_6_7;
 alias METAL_2_3 = metallib_2_3;
 alias METAL_2_4 = metallib_2_4;
 
-alias sm_2_0_GLSL_140 = sm_4_0 | glsl | spirv_1_0 | cuda | cpp;
-alias sm_2_0_GLSL_400 = sm_4_0 | glsl | spirv_1_0 | cuda | cpp;
+alias sm_2_0_GLSL_140 = _GLSL_140 + sm_4_0 | sm_4_0;
+alias sm_2_0_GLSL_400 = _GLSL_400 + sm_4_0 | sm_4_0;
```

Comment:
This doesn't seem right to me.
As discussed already, `c = (a + b) | a` is same to `c = a`

Reply:
`c = (a+b) | a` 
only if `b` is a subset of `a`, `b` implements the same stages as `a`

Reply:
@csyonghe can clarify this because I am still confused on this.
My understanding on this is something like this,
```
new_feature = cheap_gpu + expensive_gpu | cheap_gpu
```
In this expression, `new_feature` requires either `cheap_gpu and expensive_gpu` or `cheap_gpu`, which effectively same to say you don't need an `expensive_gpu`.

Reply:
In this case, _GLSL_400 + sm_4_0 | sm_4_0 will resolve to:
```
  hlsl + _sm_4_0
| glsl + glsl_spirv_1_0 + _GL_ARB_sparse_texture_clamp + _GL_EXT_samplerless_texture_functions
| glsl + _GLSL_130 + _GLSL_140 // come from the union
| spirv_1_0 + spvImageQuery + spvImageGatherExtended + spvMinLod + SPV_GOOGLE_user_type
| _cuda_sm_2_0
| metal
| cpp

```
This shouldn't be possible to express in our current data structure, right? Because we will have two disjunctions for the target `glsl`, and our data structure can only hold one disjunction at a time.

Reply:
My thinking here is we should avoid defining disjunctions in our capability definitions that would lead to disjunctions within a (target,stage) pair. In another word, we should never allow disjunctions to exists for the same target and stage.

We could just define it to be:
```
alias sm_2_0_GLSL_140 = _GLSL_140 + sm_4_0
```
So that it resolves to:
```
  hlsl + _sm_4_0  
| glsl + glsl_spirv_1_0 + _GL_ARB_sparse_texture_clamp + _GL_EXT_samplerless_texture_functions + _GLSL_130 + _GLSL_140
| spirv_1_0 + spvImageQuery + spvImageGatherExtended + spvMinLod + SPV_GOOGLE_user_type  
| _cuda_sm_2_0  
| metal  
| cpp  
```
Which is what we want.

Reply:
Ah, I am wrong, _GLSL_140 + sm_4_0 will resolve to just
```
glsl + glsl_spirv_1_0 + _GL_ARB_sparse_texture_clamp + _GL_EXT_samplerless_texture_functions + _GLSL_130 + _GLSL_140  
```

If we want the above result, it has to be
```
alias sm_2_0_GLSL_140 = (_GLSL_140 + sm_4_0 ) | (sm_4_0 + all_targets_but_glsl) 
```
where `all_targets_but_glsl = metal + hlsl + spirv + cuda + cpp`

Reply:
I think the current implementation of the CapabilitySet logic may be incorrect, since `_GLSL_140 + sm_4_0 | sm_4_0` shouldn't be a representable set.